ament/ament_lint
# Fix ament_clang_tidy

### Changes made
- The <path> argument is now meant to point towards a compile_commands.json file or a directory that contains it, instead of source files
- Unit tests source files are currently skipped because gtest has linter violations that the ROS2 package developer wouldn't have control over

### How to use/test
Follow the README to install necessary dependencies (for example on Ubuntu 18.04, do `apt install clang-tidy-6.0 clang-tools-6.0 python-yaml`).
In a workspace that contains the packages that you want to lint (or want to test the linter on):
- Build/rebuild with `colcon build --cmake-args -DCMAKE_EXPORT_COMPILE_COMMANDS=ON`
- Run `ament_clang_tidy build/`
    - Expect to see a list of linter warnings/errors printed to `stdout`
    - Prior to this fix, all of the linter errors are just complaints about missing header files
- Run `ament_clang_tidy build/ --xunit-file test.xml`
    - Expect to see a list of linter warnings/errors printed to `stdout`
    - Expect to see the same list of warnings/errors in the outputted `test.xml` file
